packages/core/src/graph/process-project-graph.ts

```diff
@@ -39,7 +38,7 @@ function visitProject(
     projectName,
     context.workspace,
     (projectConfig, dependencyName) => {
-      builder.addDependency(DependencyType.static, projectName, dependencyName);
+      builder.addImplicitDependency(projectName, dependencyName);
```

Comment:
This should use addExplicitDependency, with the csproj file that links the projects as the file parameter